libvirt/README.md

```diff
@@ -58,3 +58,229 @@ mkdir -p /opt/data-dir
+- [kubectl](https://kubernetes.io/docs/reference/kubectl/)
+- [kcli](https://kcli.readthedocs.io/en/latest/)
+
+Assume that you have a 'default' network and storage pools created in libvirtd system instance (`qemu:///system`). However,
+if you have a different pool name then the scripts should be able to handle it properly.
+
+## Create the Kubernetes cluster
+
+Use the [`kcli_cluster.sh`](./kcli_cluster.sh) script to create a simple two VMs (one master and one worker) cluster
+with the kcli tool, as:
+
+```
+./kcli_cluster.sh create
+```
+
+With `kcli_cluster.sh` you can configure the libvirt network and storage pools that the cluster VMs will be created, among
+other parameters. Run `./kcli_cluster.sh -h` to see the help for further information.
+
+If everything goes well you will be able to see the cluster running after setting your Kubernetes config with:
+
+`export KUBECONFIG=$HOME/.kcli/clusters/peer-pods/auth/kubeconfig`
+
+For example, shown below:
+
+```
+$ kcli list kube
++-----------+---------+-----------+---------------------------------------+
+|  Cluster  |   Type  |    Plan   |                  Vms                  |
++-----------+---------+-----------+---------------------------------------+
+| peer-pods | generic | peer-pods | peer-pods-master-0,peer-pods-worker-0 |
++-----------+---------+-----------+---------------------------------------+
+$ kubectl get nodes
+NAME                 STATUS   ROLES                  AGE     VERSION
+peer-pods-master-0   Ready    control-plane,master   6m8s    v1.25.3
+peer-pods-worker-0   Ready    worker                 2m47s   v1.25.3
+```
+
+## Prepare the Pod VM volume
+
+In order to build the Pod VM without installing the build tools, you should use the `docker-build` Makefile target that will
+run the entire process inside a container.
+
+Ensure that the Kata Containers repository is cloned side-by-side with the cloud-api-provider's, as both repositories will be
+mounted as volumes in the container environment. Refer to [docs/DEVELOPMENT.md](/docs/DEVELOPMENT.md) for further
+information about setup of the sources.
+
+Then run:
+
+```
+$ cd image
+$ make docker-build
+```
+
+The qcow2 image file will be created at the `output` directory as shown below:
+
+```
+$ ls output/
+podvm-7da706d-dirty-amd64.qcow2
+```
+
+Next you will need to create a volume on libvirt's system storage and upload the image content. That volume is used by
+the cloud-api-adaptor program to instantiate a new Pod VM. Still on the `image` directory, you should run `make push` just
+like below:
+
+```
+$ make push
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+100  5394  100  5394    0     0   6421      0 --:--:-- --:--:-- --:--:--  6421
+virsh -c qemu:///system vol-create-as --pool default --name podvm-base.qcow2 \
+	--capacity 107374182400 --allocation 2361393152 --prealloc-metadata \
```

Comment:
nit: change it to match the` --capacity 20G --allocation 2G `change

Reply:
done